src/AltinnCore/Runtime/Controllers/ServiceAPIController.cs

```diff
@@ -552,36 +554,54 @@ private void MapModelStateToApiResultForClient(ModelStateDictionary modelState,
                             {
-                                apiResult.ValidationResult.Warnings[modelKey].Add(ServiceTextHelper.GetServiceText(errorMesssage, serviceContext.ServiceText, null, "nb-NO"));
+                                apiResult.ValidationResult.Messages[modelKey].Warnings.Add(ServiceTextHelper.GetServiceText(errorMesssage, serviceContext.ServiceText, null, "nb-NO"));
                             }
                             else
                             {
-                                apiResult.ValidationResult.Warnings.Add(modelKey, new List<string> { ServiceTextHelper.GetServiceText(errorMesssage, serviceContext.ServiceText, null, "nb-NO") });
+                                apiResult.ValidationResult.Messages.Add(modelKey, new ApiValidationMessages
+                                {
+                                    Errors = new List<string>(),
+                                    Warnings = new List<string>
+                                    {
+                                        ServiceTextHelper.GetServiceText(errorMesssage, serviceContext.ServiceText, null, "nb-NO")
+                                    }
+                                });
                             }
                         }
                         else
                         {
-                            if (apiResult.ValidationResult.Errors.ContainsKey(modelKey))
+                            containsErrors = true;
+                            if (apiResult.ValidationResult.Messages.ContainsKey(modelKey))
                             {
-                                apiResult.ValidationResult.Errors[modelKey].Add(ServiceTextHelper.GetServiceText(error.ErrorMessage, serviceContext.ServiceText, null, "nb-NO"));
+                                apiResult.ValidationResult.Messages[modelKey].Errors.Add(ServiceTextHelper.GetServiceText(error.ErrorMessage, serviceContext.ServiceText, null, "nb-NO"));
                             }
                             else
                             {
-                                apiResult.ValidationResult.Errors.Add(modelKey, new List<string> { ServiceTextHelper.GetServiceText(error.ErrorMessage, serviceContext.ServiceText, null, "nb-NO") });
+                                apiResult.ValidationResult.Messages.Add(modelKey, new ApiValidationMessages
+                                {
+                                    Errors = new List<string>
+                                    {
+                                        ServiceTextHelper.GetServiceText(error.ErrorMessage, serviceContext.ServiceText, null, "nb-NO")
+                                    },
+                                    Warnings = new List<string>(),
+                                });
                             }
                         }
                     }
                 }
             }
 
-            if (apiResult.ValidationResult.Errors.Count > 0)
+            if (containsErrors)
```

Comment:
Errors and warnings should not be shown at the same time?

Reply:
They are shown at the same time, there is no difference here other than where the check is performed. 
If there are errors (and potentially warnings), then the user cannot move on, while if there are only warnings then the user can go to the next step